mirrord-agent/src/sniffer.rs

```diff
@@ -93,12 +93,73 @@ fn is_closed_connection(flags: u16) -> bool {
     0 != (flags & (TcpFlags::FIN | TcpFlags::RST))
 }
 
-struct ConnectionManager {
-    sessions: SessionMap,
-    index_allocator: IndexAllocator<ConnectionID>,
-    ports: HashSet<u16>,
+pub struct TcpManagerCodec {}
+
+impl PacketCodec for TcpManagerCodec {
+    type Type = Vec<u8>;
+
+    fn decode(&mut self, packet: pcap::Packet) -> Result<Self::Type, pcap::Error> {
+        Ok(packet.data.to_vec())
+    }
+}
+
+fn prepare_sniffer(interface: String) -> Result<Capture<Active>, AgentError> {
+    debug!("prepare_sniffer -> Preparing interface.");
+
+    let interface_names_match = |iface: &Device| iface.name == interface;
+    let interfaces = Device::list()?;
+
+    let interface = interfaces
+        .into_iter()
+        .find(interface_names_match)
+        .ok_or_else(|| AgentError::NotFound("Interface not found!".to_string()))?;
+
+    let mut capture = Capture::from_device(interface)?
+        .immediate_mode(true)
+        .open()?;
+
+    capture.set_datalink(Linktype::ETHERNET)?;
+    // Set a dummy filter that shouldn't capture anything. This makes the code easier.
+    capture.filter(DUMMY_BPF, true)?;
+    capture = capture.setnonblock()?;
+
+    Ok(capture)
 }
 
+struct TcpPacketData {
+    data: Vec<u8>,
```

Comment:
I would rename this to `bytes`, otherwise I foresee some `let _ = data.data.len()` call.